src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -834,7 +926,85 @@ NVTX.@annotate function set_diagnostic_edmf_precomputed_quantities_do_integral!(
                     TD.air_temperature(thermo_params, tsʲ_prev_level),
                     dt,
                 )
+            elseif moisture_model isa NonEquilMoistModel &&
+                   microphysics_model isa Microphysics2Moment
+
+                # 2M warm rain formation
+                compute_warm_precipitation_sources_2M!(
+                    Sᵖ_prev_level,
+                    Sᵖ_2_prev_level,
+                    S_n_liqʲ_prev_level,
+                    S_n_raiʲ_prev_level,
+                    S_q_liqʲ_prev_level,
+                    S_q_raiʲ_prev_level,
+                    ρʲ_prev_level,
+                    n_liqʲ_prev_level,
+                    n_raiʲ_prev_level,
+                    q_totʲ_prev_level,
+                    q_liqʲ_prev_level,
+                    q_iceʲ_prev_level,
+                    q_raiʲ_prev_level,
+                    q_snoʲ_prev_level,
+                    tsʲ_prev_level,
+                    dt,
+                    microphysics_2m_params,
+                    thermo_params,
+                )
+                # Snow not implemented
+                @. S_q_iceʲ_prev_level = FT(0)
+                @. S_q_snoʲ_prev_level = FT(0)
 
+                # Cloud sources from the updrafts
+                @. S_q_liqʲ_prev_level += cloud_sources(
+                    cloud_params.liquid,
+                    thermo_params,
+                    q_totʲ_prev_level,
+                    q_liqʲ_prev_level,
+                    q_iceʲ_prev_level,
+                    q_raiʲ_prev_level,
+                    q_snoʲ_prev_level,
+                    ρʲ_prev_level,
+                    TD.air_temperature(thermo_params, tsʲ_prev_level),
+                    dt,
+                )
+                @. S_q_iceʲ_prev_level += cloud_sources(
+                    cloud_params.ice,
+                    thermo_params,
+                    q_totʲ_prev_level,
+                    q_liqʲ_prev_level,
+                    q_iceʲ_prev_level,
+                    q_raiʲ_prev_level,
+                    q_snoʲ_prev_level,
+                    ρʲ_prev_level,
+                    TD.air_temperature(thermo_params, tsʲ_prev_level),
+                    dt,
+                )
+                # Aerosol activation
+                # TODO - right now I'm hijacking the
```

Comment:
We can read prescribed aerosol data from artifacts (Merra-2 dataset). For example the function ``microphysics_wrappers/compute_prescribed_aerosol_properties!`` reads sea salt and sulfate data and computes their number concentrations and the mean radius of seasalt. In order to use the data we need to pass a list of aerosols in the yml file.